planner/core/exhaust_physical_plans.go

```diff
@@ -2092,15 +2087,24 @@ func (p *LogicalWindow) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([
 func (p *baseLogicalPlan) exhaustPhysicalPlans(_ *property.PhysicalProperty) ([]PhysicalPlan, bool) {
 	panic("baseLogicalPlan.exhaustPhysicalPlans() should never be called.")
 }
+func (p *baseLogicalPlan) canChildPushDown() bool {
+	_, ok := p.children[0].(*DataSource)
+	// TiFlash support pushing down more operators
+	if !ok && (p.SCtx().GetSessionVars().AllowBCJ || p.SCtx().GetSessionVars().AllowMPPExecution) {
+		_, ok = p.children[0].(*LogicalProjection)
```

Comment:
switch + case is better to expand our push-down list

Reply:
updated.